feat: [003-Post-Like] 게시글 좋아요 기능 구현

jaehyeok-code opened the pull request:
Changes
---
- 게시글 좋아요 및 취소 기능 구현
- Entity, Dto, Repository를 Common모듈로 이동
- BaseEntity 클레스 제거
- JwtConfig 클래스 추가

jaehyeok-code committed:
feat: [003-Post-Like] 게시글 좋아요 기능 구현

- 게시글 좋아요 및 취소 기능 구현
- Entity, Dto, Repository를 Common모듈로 이동
- BaseEntity 클레스 제거
- JwtConfig 클래스 추가

mingming-mentor commented on common/src/main/java/com/project/common/domain/entity/User.java:
```diff
@@ -51,13 +49,19 @@ public class User extends BaseEntity {
         .password(form.getPassword())
         .nickname(form.getNickname())
         .birth(form.getBirth())
+        .createdAt(LocalDateTime.now())
+        .updatedAt(LocalDateTime.now())
```
여기도 위의 Like처럼 @PrePersist 와 @PreUpdate 를 써도 좋을 것 같아요

jaehyeok-code replied:
넵 수정해보겠습니다!! 다른부분도 확인하고 다음번 PR에 반영하겠습니다.

mingming-mentor commented on post-api/src/main/java/com/project/post/application/LikeApplication.java:
```diff
@@ -0,0 +1,22 @@
+
+  public String addLike(Long postId, Long userId) {
+    likeService.likePost(postId, userId);
+    return "좋아요가 추가되었습니다.";
```
service layer에서는 비즈니스 로직만 담고, response로 내려주는 값은 controller layer에서 만드는 것이 좋을 것 같습니다
layer 별 역할을 잘 구분지어 놓으면 추후 유지보수도 용이하고 최소한의 변경으로 구현을 할수 잇을 것 같아요

mingming-mentor commented on post-api/src/main/java/com/project/post/controller/LikeController.java:
```diff
@@ -0,0 +1,35 @@
+    return new ResponseEntity<>(result, HttpStatus.CREATED);
+  }
+
+  @DeleteMapping("/delete")
```
restful api convention이 존재합니다
이에 따르면 delete의 경우 DELETE http method를 사용하고, url에는 별도의 delete 를 넣지 않아도 될것 같아요
https://restfulapi.net/resource-naming/

mingming-mentor reviewed (approved):
수고 많으셨습니다

jaehyeok-code committed:
refactor: [003-Post-Like] API 리팩토링 및 피드백 반영

- 좋아요 기능
  - 컨트롤러 경로 단순화 (HTTP 메서드를 통한 RESTful 설계 적용)
  - LikeApplication에서 비즈니스 로직만 처리하도록 수정
- 게시글 기능
  - Post 엔티티에 @PrePersist 및 @PreUpdate 적용해 createdAt/updatedAt 자동 설정
  - PostService에서 불필요한 updatedAt 설정 코드 제거
- 전체적으로 Service와 Controller 계층의 역할 분리 및 응답 메시지 통일성 개선
